### DISCUSSION TITLE
How to keep Migrations with Model in Sync?

### DISCUSSION THREAD

**Discussion by prince-bansal** [sequelize/sequelize#17192] [Help & Questions]
Hi there, I've been using Sequelize with nodejs for a while but without migrations. Now I want to dive into migrations, I just noticed that while creating migrations we get the changes in the db only. I want to understand the flow of how to keep models and db in sync using migrations.

Suppose, we want to add a column inside an already created table, do we run migration first and then update the model manually or is there a way that can update the models automatically and keep them in sync with DB?

**Comment by kottofey** [comment]
Hi! Have you found any solution for this? I've just encountered the same problem )

**Comment by kottofey** [comment]
Orrrrrrr...... maybe i've just found a way... 

So the point is that firstly you create initial database with sequelize, describe models etc., then push your app to production. Everything works well. After a while, you decide to change something in a DB structure. To make it work and preserve all data in prod DB you need to write a migration to modify your DB structure. At the same time, you modify your code to make it corresponding to a new business logic AND a new DB structure (which you have just changed with a migration). So you actually need to run migration and push a new code simultaneously. Migration changes database structure and new code makes it working as a new logic says.

**Comment by kottofey** [comment]
Another thing I just realize, when you run migrations, the history of migrations is written to a separate table (SequelizeMeta by default or something like this). Now, even if you wipe your production database and initialize it from zero, your code would be in the latest version, and you do not need to run migrations again. Even if you try to — they won't run because migrations history table will say, all is up-to-date. This is of course if you did not touch this migrations table ) Otherwise, need to think how to workaround. Without migrations table, it might give you errors... So if you wipe your database do it by sequelize.drop() method, it will keep the migrations table.

Oh, and all migrations I did programmaticaly with umzug library, not with sequelize-cli
